src/Features/Core/Portable/CustomMessageHandler/ICustomMessageHandlerService.cs

```diff
@@ -0,0 +1,37 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Host;
+
+namespace Microsoft.CodeAnalysis.CustomMessageHandler;
+
+internal interface ICustomMessageHandlerService : IWorkspaceService
+{
+    ValueTask<RegisterHandlersResponse> LoadCustomMessageHandlersAsync(
+        string assemblyFolderPath,
+        string assemblyFileName,
+        CancellationToken cancellationToken);
```

Comment:
this does feel weird.  what would it mean to have an assemblyFileName *not* in the folder path?

Reply:
I added a check for
```
assemblyFileName.IndexOfAny(Path.GetInvalidFileNameChars()) >= 0
```
which solves the issue of `assemblyFileName` being anything else than a file name.